src/dotnet/commands/dotnet-build/CompileContext.cs

```diff
@@ -87,12 +72,35 @@ public bool Compile(bool incremental)
             return success;
         }
 
-        private bool DependencyNeedsRebuilding(ProjectContext project, ProjectDependenciesFacade dependencies)
+        private bool CompileDendencies(bool incremental)
+        {
+            if (_args.NoDependenciesValue)
+            {
+                return true;
+            }
+
+            foreach (var dependency in Sort(_rootProjectDependencies.ProjectDependenciesWithSources))
+            {
+                if (incremental && !DependencyNeedsRebuilding(ProjectContext.Create(dependency.Path, dependency.Framework)))
+                {
+                    continue;
+                }
+
+                if (!InvokeCompileOnDependency(dependency))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        private bool DependencyNeedsRebuilding(ProjectContext projectForDependency)
         {
-            return NeedsRebuilding(project, dependencies, buildOutputPath: null, intermediateOutputPath: null);
+            return NeedsRebuilding(projectForDependency, new ProjectDependenciesFacade(projectForDependency, _args.ConfigValue), buildOutputPath: null, intermediateOutputPath: null);
         }
 
-        private bool NeedsRebuilding(ProjectContext project, ProjectDependenciesFacade dependencies)
+        private bool RootProjectsNeedsRebuilding(ProjectContext project, ProjectDependenciesFacade dependencies)
```

Comment:
Good catch. The typo could be a premonition for a potential next feature when dotnet verbse receive multiple projects :)